### ISSUE TITLE
Please add option to avoid auto POST auth url for responseType other than 'token'

### ISSUE THREAD

**Issue by paresthesia69** [sahat/satellizer#548]
Hello,
I see others have requested callbacks instead of auto POST and some posting issues 404s for server urls that don't exist "auth/<provider> (404)", and the only way around this besides actually having a POST api end point to process the auth data is to use responseType: 'token'

In satellizer.js this line (476 in v0.12.4) basically allows the dev to bypass the auto auth/<provider> POST:
```
    if (defaults.responseType === 'token') {
         return oauthData;
    }
```

Can we have an optional boolean instead? Like so:
```
    if (defaults.responseType === 'token' || defaults.authPostBypass) {
         return oauthData;
    }
```

And make the authPostBypass boolean defaulted to false, so not to break existing API.  Or maybe allow for provider config.url to be null and the conditional test would be something like:
```
    if (defaults.responseType === 'token' || !!defaults.url) {
         return oauthData;
    }
```

I ask this because for some providers I'd like to have multiple response types 'code token', but not be required to support auto posting for that case.

I can create a pull request if you prefer.

**Comment by paulocheque** [comment]
+1

This would be great!

**Comment by elhamsarikhani** [comment]
+1

**Comment by tomgrooffer** [comment]
+1

**Comment by sahat** [comment]
@everyone
I just pushed an update here: https://github.com/sahat/satellizer/commit/bacd50e44b3cd0dd1d57a70dd8b1815d5cf50020. Download *satellizer.js* from the **master** branch and try it out.

Please let me know if it works, or whether it doesn't work, so I could fix it and release the new patch version.

**Comment by yahyaKacem** [comment]
hey, I downloaded satallizer.js from master as you suggested, but the work flow is not as I expected the callback url already return the token from my app's server, so all I have to do is to save it in the storage but it's not passing the token retrieved from server it give's me the url params from the call to server, the code or the access_token if I use the `responseType: 'token'`, is there's a way to get the response from the callback url sent to the parent window instead of getting the request params?
here's an example of the returned data:

    {
        code: "AQBPOVZb2mUOVUwRJh1xr_i2E2YuGGlsv374cfplQuFJ20wSxu…mPYY3nM9LFt3RIN7IZ0iAc2nQNHjN7g4MI90j8SjROFHKId-g",
        _: "_"
    }

this is from a console.log I added to satallizer.js in line: 755 before the `Popup.popupWindow.close();`.

**Comment by ngraef** [comment]
@sahat Thanks, this is exactly what I needed :+1:

**Comment by Narretz** [comment]
Was this ever fixed? I'm using 0.15.5 and it doesn't work with an OAuth2 provider. In the meantime, I am using this decorator:

```
configAndRunModule.decorator('SatellizerOAuth2', ($delegate, $q) => {
  const proto = Object.getPrototypeOf($delegate);
  const originalExchangeForToken = proto.exchangeForToken;

  // Important: Use function expression instead of fat-arrow to preseve the context (this)
  proto.exchangeForToken = function(oauth) {
    if(this.defaults.responseType === 'code' && this.defaults.noCodeExchange) {
      return $q.resolve(oauth);
    } else {
      return originalExchangeForToken.apply(this, arguments);
    }
  };

  return $delegate;
});
```

**Comment by ngraef** [comment]
@Narretz Yes, it was fixed but then got lost in the 0.15 TypeScript rewrite. PR #946 was merged in September to fix again, but there has been no release since then.
